lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -1418,6 +1423,44 @@ void TFGraphLowering::visitTFDataset(BuiltinInst *inst) {
   }
 }
 
+bool TFGraphLowering::copyGraphFunctions(const std::vector<char> &graphDefProto,
```

Comment:
It's unfortunate to serialize to graphdef then deserialize back out.  Is this the long term plan?  I'm fine with this if it is just a stepping stone.

rxwei and I had a brief conversation yesterday, I came to realize that the right long term model for us is probably to have partitioning generate a single graph per sil module, even if we have multiple start/end points scattered around the program.  Each start point would "pull" on different "calls" to TF_Functions as in your current design.

I'm heading into the office in a bit.  If you have time over GVC today or want to talk on monday, I'd love to get your thoughtso n this.

Reply:
Please add a doc comment to this function. :)

Reply:
A doc comment is available in the interface of copyGraphFunctions() -- I did not repeat it here to avoid the duplicate content drifting out of sync. :)

Agree with the waste of serializing and deserializing. Left a TODO here accordingly. The long term model sounds good. I will schedule some time tomorrow to discuss with you more (I'm tied up today).